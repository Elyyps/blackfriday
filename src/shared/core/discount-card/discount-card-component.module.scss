.discount-card {
  display: flex;
  flex-direction: column;
  border-radius: $global-border-radius;
  background: $global-background;
  color: $gray-lighter;
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.08);
  font-size: 14px;
  font-weight: 600;
  padding: 16px;
  width: 100%;
  box-sizing: border-box;
  height: 100%;
  padding-bottom: 56px;
  position: relative;
  @media (max-width: $breakpoint-xsmall-max) {
    padding-bottom: 24px;
  }
  &__body {
    flex: 1 1 auto;
  }

  &__coupon {
    box-sizing: border-box;
    font-size: 15px;
    line-height: 1.3;
    font-weight: 600;
    font-family: "Assistant", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial,
      "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
    text-align: center;
    text-decoration: none;
    display: inline-flex;
    align-items: center;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    justify-content: center;
    position: relative;
    border-radius: 2px;
    padding: 8px 30px;
    -webkit-transition: background-color 0.3s;
    transition: background-color 0.3s;
    cursor: pointer;
    text-transform: uppercase;
    border: 1px dashed #00a82f;
    background-color: white;
    color: #00a82f;
    width: 100%;
    margin-bottom: 8px;
    &:hover {
      background-color: #caf7d6;
    }
  }
  &__label {
    font-family: Assistant;
    font-size: 11px;
    font-weight: bold;
    margin-bottom: 2px;
    text-transform: uppercase;
    color: #00a82f;
  }
  &__title {
    font-family: $global-font-family-secondary;
    font-size: 24px;
    font-weight: bold;
    color: $gray-dark;
    margin: 0;
  }
  &__subtitle {
    margin-top: 24px;
    @media (max-width: $breakpoint-xsmall-max) {
      display: none;
    }
  }

  &__image {
    margin-bottom: 16px;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 85px;

    img,
    svg {
      margin: 0 auto;
      max-height: 100%;
    }
  }

  &__content {
    margin-bottom: 8px;
    font-size: 14px;
    text-align: center;
    @media (max-width: $breakpoint-xsmall-max) {
      text-align: left;
      margin-bottom: 16px;
    }

    a {
      color: $gray;
      text-decoration: none;
    }
  }

  &__action {
    flex: 0 0 auto;
    > * {
      width: 100%;
    }
  }

  &__status {
    color: $gray-lighter;
    margin-bottom: 8px;

    &-title {
      text-transform: uppercase;
      font-size: 11px;
      font-weight: bold;
    }
    &-bar {
      position: relative;
      height: 3px;
      background: $tint-gray-light;
      border-radius: 3px;
      overflow: hidden;
    }
    &-bar-filled {
      position: absolute;
      left: 0;
      top: 0;
      height: 3px;
      background: $tint-gray-light;
    }
    &.green {
      color: $brand-secondary;
      .discount-card__status-bar-filled {
        background: $brand-secondary;
      }
    }
    &.orange {
      color: #ff9d0a;
      .discount-card__status-bar-filled {
        background: #ff9d0a;
      }
    }
  }

  &--responsive {
    .discount-card {
      &__body {
        display: flex;
        flex-wrap: wrap;
      }

      &__image {
        order: 2;
        width: 90px;
        height: auto;
        flex: 0 0 90px;
        margin-bottom: 0;
      }

      &__content {
        order: 1;
        flex: 1 1 60%;
        margin-bottom: 0;
      }

      &__action {
        width: 50%;
        margin-left: auto;
      }

      &__status {
        order: 3;
        width: 100%;
        clear: both;
        margin-top: 16px;
        margin-bottom: 16px;
        position: relative;
        &-title {
          position: absolute;
          left: 0;
          top: 28px;
          width: 50%;
        }
      }
    }
  }
  &--modify {
    position: relative;
    &:after {
      content: "";
      position: absolute;
      left: 0;
      top: 0;
      width: 56px;
      height: 56px;
      border-bottom-right-radius: 100%;
      background-color: $bg-muted;
      background-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgaGVpZ2h0PSIxNC4wMyIgdmlld0JveD0iMCAwIDE2IDE0LjAzIj48dGl0bGU+aGVhcnQtZmlsbGVkPC90aXRsZT48cGF0aCBkPSJNMTQuNzI5LDEuMzQ4QTQuNDYsNC40NiwwLDAsMCw5LjguMzUxYTUuNDE0LDUuNDE0LDAsMCwwLTEuNzYyLDEuNEE1LjUsNS41LDAsMCwwLDYuMjc3LjM1MWE0LjQ2MSw0LjQ2MSwwLDAsMC00LjkyOSwxLDQuNDYxLDQuNDYxLDAsMCwwLDAsNi4zOWw2LjY5LDYuMjkyLDYuNjkxLTYuM2E0LjY0Niw0LjY0NiwwLDAsMCwwLTYuMzg2WiIgZmlsbD0iIzc3Nzc3NyIvPjwvc3ZnPg==);
      background-repeat: no-repeat;
      background-position: 16px 18px;
    }
    .discount-card {
      &__content {
        text-align: center;
        order: 0;
      }
      &__status {
        margin-bottom: 16px;
      }
      &__title {
        font-size: 24px;
        margin-bottom: 24px;
      }
      &__body {
        align-items: center;
        flex-direction: column;
      }
      &__image {
        order: 0;
      }
      &__status-title {
        position: static;
      }
      &__action {
        @media (max-width: $breakpoint-small-max) {
          margin: 0 auto;
        }
      }
    }
  }
}
