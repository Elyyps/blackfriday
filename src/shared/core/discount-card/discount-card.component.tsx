import * as React from "react";
import styles from "./discount-card-component.module.scss";
import { ImageComponent, Button } from "@app/prep/modules-prep/core";
import IconDefault from "@assets/icons/link.svg";
import { TimeLeftBarComponent } from "../time-left-bar";
import { injectIntl, InjectedIntlProps } from "react-intl";

interface IDiscountCardComponentProps {
  buttonLink: string;
  buttonText: string;
  content: string;
  coupon: string;
  image: string;
  range: number;
  status: string;
  title: string;
}

const component = (props: IDiscountCardComponentProps & InjectedIntlProps) => {
  const { title, status, content, image, range, buttonText, buttonLink, coupon } = props;
  const [messageDisplayed, setMessageDisplayed] = React.useState<boolean>(false);
  const [couponCopied, setCouponCopied] = React.useState<boolean>(false);

  const ClipLoaderTimer = 1500;

  const [] = React.useState<boolean>(false);
  const statusBar = (rangeNumber: any) => {
    const limit = 0.5;
    if (rangeNumber > limit) {
      return "#00a82f";
    }

    if (rangeNumber === "0") {
      return "none";
    }

    if (rangeNumber <= limit) {
      return "orange";
    }
  };
  const displayMessage = () => {
    if (!couponCopied) {
      setMessageDisplayed(true);
      setCouponCopied(true);

      setTimeout(() => {
        setMessageDisplayed(false);
      }, ClipLoaderTimer);
    }
  };

  return (
    <div className={styles["discount-card"]}>
      <div className={styles["discount-card__body"]}>
        <div className={styles["discount-card__image"]}>
          <ImageComponent alt="Logo" src={image} />
        </div>
        <div style={{ color: statusBar(range) }} className={styles["discount-card__label"]}>
          {status}
        </div>
        <TimeLeftBarComponent color={statusBar(range)} range={range} />
        <div className={styles["discount-card__content"]}>
          <div className={styles["discount-card__title"]}>{title}</div>
          <div className={`${styles["discount-card__subtitle"]}  uk-visible@s`}>{content}</div>
        </div>
      </div>
      <div className={styles["discount-card__action"]}>
        <button
          onClick={displayMessage}
          className={styles["discount-card__coupon"]}
          style={couponCopied ? { backgroundColor: "#caf7d6" } : { backgroundColor: "white" }}
        >
          {coupon}
        </button>
        {messageDisplayed && (
          <div className={styles["discount-card__tooltip"]} aria-hidden="true">
            <div className={styles["discount-card__tooltip__message"]}>
              {props.intl.formatMessage({ id: "discount-code" })}
            </div>
            <div className={styles["discount-card__tooltip__arrow"]} />
          </div>
        )}
        <Button title={buttonText} href={buttonLink} variant={"primary"} icon={IconDefault} />
      </div>
    </div>
  );
};
const DiscountCardComponent = injectIntl(component);
export { DiscountCardComponent };
