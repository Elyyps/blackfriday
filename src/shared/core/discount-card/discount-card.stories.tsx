import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { DiscountCardComponent } from "./discount-card.component";
import CoolBlue from "@assets/cool-blue.svg";

storiesOf("DiscountCard", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <DiscountCardComponent
      content="Content"
      range={1}
      status="status"
      title="Title"
      image={CoolBlue}
      coupon="COUPON"
      buttonLink="/"
      buttonText=""
    />
  ));
