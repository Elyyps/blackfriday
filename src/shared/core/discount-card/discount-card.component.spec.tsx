
import React from "react";
import { shallow } from "enzyme";
import { DiscountCardComponent } from "./discount-card.component";

describe("[DiscountCard]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <DiscountCardComponent />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});

  