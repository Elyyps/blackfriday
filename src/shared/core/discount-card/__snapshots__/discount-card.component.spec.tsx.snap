// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[DiscountCard] should render component without crashing 1`] = `
<div
  className="discount-card"
>
  <div
    className="discount-card__body"
  >
    <div
      className="discount-card__image"
    >
      <ImageComponent
        src="cool-blue.svg"
      />
    </div>
    <div
      className="discount-card__label"
      style={
        Object {
          "color": "#00a82f",
        }
      }
    >
      status
    </div>
    <TimeLeftBarComponent
      color="#00a82f"
      range={1}
    />
    <div
      className="discount-card__content"
    >
      <div
        className="discount-card__title"
      >
        Title
      </div>
      <div
        className="discount-card__subtitle  uk-visible@s"
      >
        Content
      </div>
    </div>
  </div>
  <div
    className="discount-card__action"
  >
    <button
      className="discount-card__coupon"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      COUPON
    </button>
    <Button
      href="/"
      icon="link.svg"
      title=""
      variant="primary"
    />
  </div>
</div>
`;
